vault/data_source_namespace_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+func TestAccDataSourceNamespace(t *testing.T) {
+	testutil.SkipTestAccEnt(t)
+
+	path := acctest.RandomWithPrefix("tf-ns")
+	pathChild := acctest.RandomWithPrefix("tf-child")
+
+	client := testProvider.Meta().(*provider.ProviderMeta).GetClient()
+	providerNS := util.TrimSlashes(client.Namespace())
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testNamespaceDestroy(path),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceNamespaceConfig_nested(path, pathChild),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.vault_namespace.parent", consts.FieldPath, path),
+					resource.TestCheckResourceAttr("data.vault_namespace.parent", consts.FieldPathFQ, path),
+					resource.TestCheckResourceAttr("data.vault_namespace.parent", consts.FieldID, fmt.Sprintf("%s/%s/", providerNS, path)),
+					resource.TestCheckResourceAttrSet("data.vault_namespace.parent", consts.FieldNamespaceID),
+
+					resource.TestCheckResourceAttr("data.vault_namespace.child", consts.FieldPath, pathChild),
+					resource.TestCheckResourceAttr("data.vault_namespace.child", consts.FieldPathFQ, fmt.Sprintf("%s/%s", path, pathChild)),
+					resource.TestCheckResourceAttr("data.vault_namespace.child", consts.FieldID, fmt.Sprintf("%s/%s/%s/", providerNS, path, pathChild)),
+					resource.TestCheckResourceAttrSet("data.vault_namespace.child", consts.FieldNamespaceID),
+				),
+			},
+		},
+	})
+
+	/* TODO: Enable when #1766 merged
```

Comment:
Maybe this can be enabled now that https://github.com/hashicorp/terraform-provider-vault/pull/2033 is merged?